### ISSUE TITLE
Most used configuration options

### ISSUE THREAD

**Issue by blyxyas** [rust-lang/rust-clippy#15821]
Note that lots of these were used by the same 7-8 repositories, because there are a handful of repos that like to declare every configuration option even when they're set to the default (like [wgsl-analyzer](https://github.com/wgsl-analyzer/wgsl-analyzer))

Even worse, there are configuration option that the only repo using them is the `rust-clippy` one or that no one is using an option that isn't the default. We should think if we want to remove them or think about improving discoverability of our configuration options)

|**Configuration option**| **Number of Github search results (excl. forks)**|
|--------------------------------|---|
|`absolute-paths-allowed-crates`|47|
|`absolute-paths-max-segments`|32|
|`accept-comment-above-attributes`|19|
|`accept-comment-above-statement`|19|
|`allow-comparison-to-zero`|10|
|`allow-dbg-in-tests`|215|
|`allow-exact-repetitions`|4|
|`allow-expect-in-consts`|8|
|`allow-expect-in-tests`|380|
|`allow-indexing-slicing-in-tests`|32|
|`allow-mixed-uninlined-format-args`|51|
|`allow-one-hash-in-raw-strings`|7|
|`allow-panic-in-tests`|142|
|`allow-print-in-tests`|151|
|`allow-private-module-inception`|35|
|`allow-unwrap-in-consts`|8|
|`allow-unwrap-in-tests`|540|
|`allow-useless-vec-in-tests`|8|
|`allowed-dotfiles`|8|
|`allowed-duplicate-crates`|101|
|`allowed-prefixes`|16|
|`allowed-scripts`|30|
|`allowed-wildcard-imports`|41|
|`arithmetic-side-effects-allowed`|41|
|`arithmetic-side-effects-allowed-binary`|11|
|`arithmetic-side-effects-allowed-unary`|11|
|`array-size-threshold`|72|
|`avoid-breaking-exported-api`|422|
|`await-holding-invalid-types`|46|
|`blacklisted-names`|68|
|`cargo-ignore-publish`|12|
|`check-incompatible-msrv-in-tests`|7|
|`check-inconsistent-struct-field-initializers`|11|
|`check-private-items`|132|
|`cognitive-complexity-threshold`|528|
|`const-literal-digits-threshold`|1|
|`cyclomatic-complexity-threshold`|54|
|`disallowed-macros`|252|
|`disallowed-methods`|780|
|`disallowed-names`|192|
|`disallowed-types`|548|
|`doc-valid-idents`|628|
|`enable-raw-pointer-heuristic-for-send`|9|
|`enforce-iter-loop-reborrow`|15|
|`enforced-import-renames`|40|
|`enum-variant-name-threshold`|114|
|`enum-variant-size-threshold`|190|
|`excessive-nesting-threshold`|157|
|`future-size-threshold`|63|
|`ignore-interior-mutability`|111|
|`large-error-threshold`|122|
|`lint-commented-code`|13|
|`lint-inconsistent-struct-field-initializers`|5|
|`literal-representation-threshold`|57|
|`matches-for-let-else`|14|
|`max-fn-params-bools`|168|
|`max-include-file-size`|115|
|`max-struct-bools`|65|
|`max-suggested-slice-pattern-length`|31|
|`max-trait-bounds`|53|
|`min-ident-chars-threshold`|19|
|`missing-docs-allow-unused`|7|
|`missing-docs-in-crate-items`|120|
|`module-item-order-groupings`|11|
|`msrv`|876|
|`pass-by-value-size-limit`|77|
|`pub-underscore-fields-behavior`|7|
|`semicolon-inside-block-ignore-singleline`|21|
|`semicolon-outside-block-ignore-multiline`|31|
|`single-char-binding-names-threshold`|176|
|`source-item-ordering`|24|
|`stack-size-threshold`|98|
|`standard-macro-braces`|63|
|`struct-field-name-threshold`|15|
|`suppress-restriction-lint-in-const`|25|
|`too-large-for-stack`|69|
|`too-many-arguments-threshold`|628|
|`too-many-lines-threshold`|366|
|`trait-assoc-item-kinds-order`|10|
|`trivial-copy-size-limit`|194|
|`type-complexity-threshold`|600|
|`unnecessary-box-size`|12|
|`unreadable-literal-lint-fractions`|20|
|`upper-case-acronyms-aggressive`|84|
|`vec-box-size-threshold`|72|
|`verbose-bit-mask-threshold`|41|
|`warn-on-all-wildcard-imports`|142|
|`warn-unsafe-macro-metavars-in-private-macros`|6|

**Comment by blyxyas** [comment]
Also, maybe we could look into removing some of these, because for example `const-literal-digits-threshold` has a single project on Github that uses it and there are a handful with 5-6 projects using them. Maybe they're heavily used by more corporate projects that are not on Github, but having more features also means more surface for bugs

**Comment by TomFryersMidsummer** [comment]
We use two of the rarely used settings across our 28 000 lines of Rust, so maybe can offer some insight into why people might be using them.

### `unreadable-literal-lint-fractions = false`
Our codebase is pretty floating-point heavy. Almost all of our floating point literals fall into two categories.

1. Simple values, like `1.0` or `-0.5`. For these, the lint doesn't do anything, because there aren't enough decimal places to trigger it.
2. Values pasted from somewhere, such as a high-precision calculation outside of Rust, or `Debug` output of case exhibiting a bug. In these cases, readability is secondary to terseness, as the value is more-or-less opaque, regardless of how it's displayed.

I am also not used to reading post-decimal-point digit grouping, as it's pretty unusual, so wouldn't get much of a readability benefit in any case.

### `semicolon-inside-block-ignore-singleline = true`

Rustfmt does this:
```rust
fn main() {
    unsafe { x.cause_danger() };
    unsafe {
        x.cause_danger();
    }
    unsafe {
        x.cause_lots_and_lots_and_lots_and_lots_and_lots_and_lots_and_lots_and_lots_of_danger()
    };
    unsafe {
        x.cause_lots_and_lots_and_lots_and_lots_and_lots_and_lots_and_lots_and_lots_of_danger();
    }
}
```
and I think that the first and last cases look nicest.

<details>
<summary>Our full <code>.clippy.toml</code>, in case it's of interest</summary>

```toml
doc-valid-idents = ["GeoTIFF", "GeoJSON", "XLink", ".."]
avoid-breaking-exported-api = false
excessive-nesting-threshold = 6
unreadable-literal-lint-fractions = false
disallowed-methods = [
    { path = "f32::round", reason = "round_ties_even is less biased and faster" },
    { path = "f64::round", reason = "round_ties_even is less biased and faster" },
]
disallowed-types = [
    { path = "std::num::NonZeroI8", replacement = "std::num::NonZero<i8>" },
    { path = "std::num::NonZeroI16", replacement = "std::num::NonZero<i16>" },
    { path = "std::num::NonZeroI32", replacement = "std::num::NonZero<i32>" },
    { path = "std::num::NonZeroI64", replacement = "std::num::NonZero<i64>" },
    { path = "std::num::NonZeroI128", replacement = "std::num::NonZero<i128>" },
    { path = "std::num::NonZeroIsize", replacement = "std::num::NonZero<isize>" },
    { path = "std::num::NonZeroU8", replacement = "std::num::NonZero<u8>" },
    { path = "std::num::NonZeroU16", replacement = "std::num::NonZero<u16>" },
    { path = "std::num::NonZeroU32", replacement = "std::num::NonZero<u32>" },
    { path = "std::num::NonZeroU64", replacement = "std::num::NonZero<u64>" },
    { path = "std::num::NonZeroU128", replacement = "std::num::NonZero<u128>" },
    { path = "std::num::NonZeroUsize", replacement = "std::num::NonZero<usize>" },
]
semicolon-inside-block-ignore-singleline = true
```
</details>
